### ISSUE TITLE
GetTextReader() in SagaPersister with Npgsql 4.0.0 throws exception

### ISSUE THREAD

**Issue by jenspettersson** [Particular/NServiceBus.Persistence.Sql#300]
In Npgsql `4.0.0`, it's no longer possible to do more than one `GetTextReader(...)` in the same data reader. See https://github.com/npgsql/npgsql/issues/1986 - apparently, it's by design. 

This causes the SagaPersister, when using PostgreSQL, to throw on https://github.com/Particular/NServiceBus.Persistence.Sql/blob/develop/src/SqlPersistence/Saga/SagaPersister_Get.cs#L94 as the stream is closed after the first GetTextReader is used.

We're using NSB Persistence along with Npgsql.EntityFrameworkCore.PostgreSQL and the `2.1.0`is using `Npgsql 4.0.0` and we noticed this behavior when upgrading from `2.1.0-preview1` that uses `Npgsql 3.2.7`.

**Comment by SzymonPobiega** [comment]
@jenspettersson reading the comment here https://github.com/npgsql/npgsql/issues/1986#issuecomment-395834333 and the release notes http://www.npgsql.org/doc/release-notes/4.0.html I understand that:
 * It is not supported to have multiple stream readers open *concurrently*
 * It is still supported to have multiple stream readers open *sequentially* on a single reader

Looking at the `SagaPersister` code around line 94 I see that the `ReadMetadata` method opens and closes the stream reader so it should be OK to open a new one for the body. I'll ask the npgsql guys to clarify.

**Comment by weralabaj** [comment]
@jenspettersson Could you also clarify which version of SQL persistence are you using?

**Comment by jenspettersson** [comment]
@weralabaj NServiceBus.Persistence.Sql 4.1.1

**Comment by jenspettersson** [comment]
The Npgsql-guys now tagged it as a bug at their end, so don’t invest too much time in this issue at the moment.

**Comment by SzymonPobiega** [comment]
@jenspettersson 👍 It seems like a bug on their side.

**Comment by jenspettersson** [comment]
FYI: https://github.com/npgsql/npgsql/issues/1986#issuecomment-396760096

I've tested the Saga persister with `npgsql 4.1.0-ci.1042` from their unstable myget feed and it now works as expected.

Feel free to close this issue, unless you're planning to do anything else first. Maybe a comment somewhere about issues when using the 4.0.0 npgsql version? Your samples only uses `3.*` npgsql version, so you're safe there, but for us that also uses the Npgsql entity framework we got upgraded to 4.0.0 after upgrading the ef package to 2.1, so it happened "behind the scenes" and took a while to track down :)

Anyway, thanks for the help.
